### PR TITLE
encrypt the keyPair in memory using the encryption key

### PR DESCRIPTION
**Pull request by nandubatchu**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -484,4 +483,29 @@ export class CruxClient {
...
+
+    private _encryptKeyPair = async () => {
+        // assuming that the decrypted keyPair is stored as an object
+        if (typeof this._keyPair === "object") {
+            this._keyPair = JSON.stringify(await this._encryption.encryptJSON(this._keyPair, await this._getEncryptionKey()));
+        }
+    }
+
+    private _decryptKeyPair = async () => {
+        // assuming that the encrypted keyPair is stored as an object
```
Comment (govind1246) [inline src/index.ts]: comment and code aren't conveying the same behavior.

File: src/index.ts [hunk 2]
```diff
@@ -145,42 +145,35 @@ export class CruxClient {
...
-            this._authenticated = false;
-        }
+        this._encryption = options.encryption || encryption.Encryption;
+        this._nameService = options.nameService;
+        this.walletClientName = options.walletClientName;
 
-        this._encryption = this._options.encryption || encryption.Encryption;
-        this._nameService = this._options.nameService;
-        this.walletClientName = this._options.walletClientName;
+        this._setKeyPairPromise = this._setKeyPair(options.privateKey);
```
Comment (govind1246) [inline src/index.ts]: are we cleaning privateKey from options once it's assigned to keyPair?

File: src/index.ts [hunk 3]
```diff
@@ -306,13 +299,16 @@ export class CruxClient {
                     throw errors.ErrorHelper.getPackageError(null, errors.PackageErrorCode.ExistingCruxIDFound, this._payIDClaim.virtualAddress);
                 }
 
-                let identityClaim: nameService.IIdentityClaim;
+                let identityClaim: nameService.IIdentityClaim | undefined;
+                await this._decryptKeyPair();
                 identityClaim = {secrets: {identityKeyPair: this._keyPair}};
```
Comment (govind1246) [inline src/index.ts]: what's the difference b/w identityClaim and this._keyPair. As it creating two variables with the private key in memory.

File: src/index.ts (hunk 1 above)
Comment (nandubatchu) [reply src/index.ts]: removed them

File: src/index.ts (hunk 2 above)
Comment (nandubatchu) [reply src/index.ts]: removing post call to the setter method

File: src/index.ts (hunk 3 above)
Comment (nandubatchu) [reply src/index.ts]: there is just a difference of object interface.

**Comment by nandubatchu** [comment]
https://stackoverflow.com/questions/684575/how-to-quickly-clear-a-javascript-object

**Pull request merged**